substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/ReflectionPlugins.java

```diff
@@ -267,6 +278,58 @@ private static boolean processGetField(GraphBuilderContext b, ResolvedJavaMethod
         return false;
     }
 
+    private static boolean processGetFields(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver,
+                                           SnippetReflectionProvider snippetReflection, boolean declared, boolean analysis, boolean hosted) {
+        if (receiver.isConstant()) {
+            Class<?> clazz = snippetReflection.asObject(Class.class, receiver.get().asJavaConstant());
+
+            String target = clazz.getTypeName() + ".getDeclaredFields()";
+            try {
+                Field[] fields = declared ? clazz.getDeclaredFields() : clazz.getFields();
+                if (fields.length == 0) {
```

Comment:
If the class doesn't declare any fields `getDeclaredFields()` could still be intrinsified to return an empty array.